File: db4oj/core/src/com/db4o/internal/btree/BTreeNodeCacheEntry.java
/* Copyright (C) 2004 - 2010  Versant Inc.  http://www.db4o.com */

package com.db4o.internal.btree;

import com.db4o.internal.*;

/**
 * @exclude
 */
public class BTreeNodeCacheEntry {
	
	public final BTreeNode _node;
	
	private ByteArrayBuffer _buffer;

	public BTreeNodeCacheEntry(BTreeNode node) {
		_node = node;
	}
	
	public ByteArrayBuffer buffer() {
		return _buffer;
	}

	public void buffer(ByteArrayBuffer buffer) {
		_buffer = buffer;
	}

}

File: db4o.net/Db4objects.Db4o/Db4objects.Db4o/Internal/Btree/BTreeNodeCacheEntry.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

using Db4objects.Db4o.Internal;
using Db4objects.Db4o.Internal.Btree;

namespace Db4objects.Db4o.Internal.Btree
{
	/// <exclude></exclude>
	public class BTreeNodeCacheEntry
	{
		public readonly BTreeNode _node;

		private ByteArrayBuffer _buffer;

		public BTreeNodeCacheEntry(BTreeNode node)
		{
			_node = node;
		}

		public virtual ByteArrayBuffer Buffer()
		{
			return _buffer;
		}

		public virtual void Buffer(ByteArrayBuffer buffer)
		{
			_buffer = buffer;
		}
	}
}
